serialize/unserialize.go

```diff
@@ -203,8 +205,10 @@ func unMarshalArray(reader *bytes.Reader) (interface{}, error) {
 
 	if indexLen == arrLen {
 		var slice []interface{}
-		for _, row := range val {
-			slice = append(slice, row)
+		// fix map sort
+		for i := 0; i < len(val); i++ {
+			k, _ := utils.NumericalToString(i)
+			slice = append(slice, val[k])
```

Comment:
@crazykun  This needs to deal with the case where the index is not a numeric type.